modules/serverless-invoking-quarkus-functions.adoc

```diff
@@ -0,0 +1,157 @@
+// Module included in the following assemblies
+//
+// * /serverless/functions/serverless-developing-quarkus-functions.adoc
+
+[id="serverless-invoking-quarkus-functions_{context}"]
+= About invoking Quarkus functions
+
+You can create a Quarkus project that responds to CloudEvents, or one that responds to simple HTTP requests. CloudEvents in Knative are transported over HTTP as a POST request, so either function type can listen and respond to incoming HTTP events.
+
+When an incoming request is received, Quarkus functions are invoked with an instance of a permitted type.
+
+.Function invocation options
+[options="header",cols="d,d,m"]
+|====
+|Invocation method |Data type contained in the instance |Example of data
+|HTTP POST request | JSON object in the body of the request |`{ "customerId": "0123456", "productId": "6543210" }`
+|HTTP GET request | Data in the query string |`?customerId=0123456&productId=6543210`
+|`CloudEvent` | JSON object in the `data` property |`{ "customerId": "0123456", "productId": "6543210" }`
+|====
+
+The following example shows a function that receives and processes the `customerId` and `productId` purchase data listed in the previous table:
+
+.Example Quarkus function
+[source,java]
+----
+public class Functions {
+    @Funq
+    public void processPurchase(Purchase purchase) {
+        // process the purchase
+    }
+}
+----
+
+The corresponding `Purchase` JavaBean class that contains the purchase data looks as follows:
+
+.Example class
+[source,java]
+----
+public class Purchase {
+    private long customerId;
+    private long productId;
+    // getters and setters
+}
+----
+
+[id="serverless-invoking-quarkus-functions-examples_{context}"]
+== Invocation examples
+
+The following example code defines three functions named `withBeans`, `withCloudEvent`, and `withBinary`;
+
+.Example
+[source,java]
+----
+import io.quarkus.funqy.Funq;
+import io.quarkus.funqy.knative.events.CloudEvent;
+
+public class Input {
+    private String message;
+
+    // getters and setters
+}
+
+public class Output {
+    private String message;
+
+    // getters and setters
+}
+
+public class Functions {
+    @Funq
+    public Output withBeans(Input in) {
+        // function body
+    }
+
+    @Funq
+    public CloudEvent<Output> withCloudEvent(CloudEvent<Input> in) {
+        // function body
+    }
+
+    @Funq
+    public void withBinary(byte[] in) {
+        // function body
+    }
+}
+----
+
+The `withBeans` function of the `Functions` class can be invoked by:
+
+* An HTTP POST request with JSON body:
```

Comment:
s/with JSON body/with a JSON body (or "the"? not sure, but seems it should have an article)